eth/skipped_txn_filter.go

```diff
@@ -0,0 +1,41 @@
+package eth
+
+import (
+	"context"
+
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/eth/tracers"
+	"github.com/scroll-tech/go-ethereum/rollup/circuitcapacitychecker"
+	"github.com/scroll-tech/go-ethereum/rpc"
+)
+
+type NewCircuitCapacityChecker func() *circuitcapacitychecker.CircuitCapacityChecker
+
+type CurrentHeaderReader interface {
+	CurrentHeader() *types.Header
+}
+
+type SkippedTxnFilter struct {
+	headReader                CurrentHeaderReader
+	tracer                    tracers.TraceBlock
+	newCircuitCapacityChecker NewCircuitCapacityChecker
+}
+
+func NewSkippedTxnFilter(headReader CurrentHeaderReader, tracer tracers.TraceBlock, newCircuitCapacityChecker NewCircuitCapacityChecker) *SkippedTxnFilter {
+	return &SkippedTxnFilter{
+		headReader:                headReader,
+		tracer:                    tracer,
+		newCircuitCapacityChecker: newCircuitCapacityChecker,
+	}
+}
+
+func (f *SkippedTxnFilter) Apply(txn *types.Transaction, local bool) error {
+	// always execute on the latest block
+	blockToExecuteOn := rpc.BlockNumberOrHashWithHash(f.headReader.CurrentHeader().Hash(), false)
```

Comment:
This doesn't really work. Because if the incoming txn depends on a pending txn in the pool (which is a really common scenario, for example; due to nonce), it's not possible to the trace it.